src/System.Numerics.Vectors/tests/GenericVectorTests.netcoreapp.cs

```diff
@@ -212,15 +284,121 @@ public partial class GenericVectorTests
         [Fact]
         public void ConstructorWithUnsupportedTypes_DateTime() => TestConstructorWithUnsupportedTypes<DateTime>();
         [Fact]
-        public void ConstructorWithUnsupportedTypes_Char() => TestConstructorWithUnsupportedTypes<char>();
+        public void ConstructorWithUnsupportedTypes_Char() => TestConstructorWithUnsupportedTypes<Char>();
 
         private void TestConstructorWithUnsupportedTypes<T>() where T : struct
         {
+            Assert.Throws<NotSupportedException>(() => new Vector<T>(new ReadOnlySpan<byte>(new byte[4])));
+            Assert.Throws<NotSupportedException>(() => new Vector<T>(new ReadOnlySpan<T>(new T[4])));
             Assert.Throws<NotSupportedException>(() => new Vector<T>(new Span<T>(new T[4])));
         }
 
         #endregion Tests for constructors using unsupported types
 
         #endregion Constructor Tests
+
+        #region CopyTo (span) Tests
+        [Fact]
+        public void CopyToSpanByte() { TestCopyToSpan<byte>(); }
+        [Fact]
+        public void CopyToSpanSByte() { TestCopyToSpan<sbyte>(); }
+        [Fact]
+        public void CopyToSpanUInt16() { TestCopyToSpan<ushort>(); }
+        [Fact]
+        public void CopyToSpanInt16() { TestCopyToSpan<short>(); }
+        [Fact]
+        public void CopyToSpanUInt32() { TestCopyToSpan<uint>(); }
+        [Fact]
+        public void CopyToSpanInt32() { TestCopyToSpan<int>(); }
+        [Fact]
+        public void CopyToSpanUInt64() { TestCopyToSpan<ulong>(); }
+        [Fact]
+        public void CopyToSpanInt64() { TestCopyToSpan<long>(); }
+        [Fact]
+        public void CopyToSpanSingle() { TestCopyToSpan<float>(); }
+        [Fact]
+        public void CopyToSpanDouble() { TestCopyToSpan<double>(); }
+        private void TestCopyToSpan<T>() where T : struct
+        {
+            var initialValues = GenerateRandomValuesForVector<T>();
```

Comment:
Nit: var => whatever type it actually is

Reply:
I was just talking to @terrajobst about this. @terrajobst, see the use of `var` here, which increases the code review difficulty. If you had to guess, from local context, what type do you think is `initialValues`?

<details>
  <summary>Answer:</summary>

https://github.com/dotnet/corefx/blob/82c6f6320aeb14186f0438e77bfdc383eb631d05/src/System.Numerics.Vectors/tests/Performance/Constructor/GenericVectorConstructorTests.tt#L88

</details>

Reply:
I wanted to change as little as possible in the existing code, even if it meant propagating undesirable coding conventions. If this is causing substantial agitation for reviewers I can change the .tt file all the way up and down as part of this PR. :)

Reply:
> If this is causing substantial agitation for reviewers I can change the .tt file all the way up and down as part of this PR. 

Yes please :)  But since this code in general predated your PR (right?), it can be a separate PR to just fix that up.